packages/babel-plugin-transform-block-scoping/src/index.ts

```diff
@@ -290,14 +324,17 @@ const loopVisitor = {
     path.skip();
   },
 
-  "BreakStatement|ContinueStatement|ReturnStatement"(path, state) {
+  "BreakStatement|ContinueStatement|ReturnStatement"(
+    path: NodePath<t.BreakStatement | t.ContinueStatement | t.ReturnStatement>,
+    state,
+  ) {
     const { node, scope } = path;
     if (node[this.LOOP_IGNORE]) return;
 
     let replace;
     let loopText = loopNodeTo(node);
 
-    if (loopText) {
+    if (loopText && !t.isReturnStatement(node)) {
```

Comment:
Is this change just to infer better types? If it is, it would be better to do this:
```suggestion
    if (loopText) {
      if (t.isReturnStatement(node)) throw new Error("Internal error: unexpected return statement with `loopText`");
```